src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -4927,22 +4927,26 @@ void MacroAssembler::verified_entry(int framesize, int stack_bang_size, bool fp_
   }
 }
 
-// clear memory of size 'cnt' qwords, starting at 'base' using XMM/YMM registers
-void MacroAssembler::xmm_clear_mem(Register base, Register cnt, XMMRegister xtmp) {
+#if COMPILER2_OR_JVMCI
+
+// clear memory of size 'cnt' qwords, starting at 'base' using XMM/YMM/ZMM registers
+void MacroAssembler::xmm_clear_mem(Register base, Register cnt, Register rtmp, XMMRegister xtmp) {
   // cnt - number of qwords (8-byte words).
   // base - start address, qword aligned.
   Label L_zero_64_bytes, L_loop, L_sloop, L_tail, L_end;
-  if (UseAVX >= 2) {
+  bool use64byteVector = MaxVectorSize == 64 && AVX3Threshold == 0;
```

Comment:
The comment for `AVX3Threshold` says:
```
  "Minimum array size in bytes to use AVX512 intrinsics"
  "for copy, inflate and fill. When this value is set as zero"
  "compare operations can also use AVX512 intrinsics.")
```

Should we mention clear memory there as well?

Reply:
Hi Nils, comments already cover "fill", clear memory is semantically equivalent to fill by zero.

Reply:
Okay, that's fine.